spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteManifestsProcedure.java

```diff
@@ -75,6 +90,57 @@ public void testRewriteLargeManifests() {
         "Must have 4 manifests", 4, table.currentSnapshot().allManifests(table.io()).size());
   }
 
+  @Test
+  public void testRewriteLargeManifestsWithDatePartitionedTables() {
```

Comment:
Nice PR. 

Can we please also add a timestamp testcase as the code modification touches that area? (maybe the same test case with parameterised inputs)?

Reply:
added an UT for timestamp as well.